[Term Entry] C++ Math Functions: `pow()` (#1000)
Codecademy/docs: content/cpp/concepts/math-functions/terms/pow/pow.md

---
Title: 'pow()'
Description: 'Returns the result of a base raised to the power of an exponent.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

> The **`pow()`** function returns the result of a base raised to the power of an exponent.

Reviewer: ```suggestion
The **`pow()`** function returns the result of a base value raised to the power of an exponent.
```

## Syntax

```cpp
pow(base, exponent)
```

> 

Reviewer: Can we put some text below the syntax snippet that describes the domain and range of `base` and `exponent` as well as what errors are caused by which invalid input(s)?

## Example

The following example uses the `pow()` function to raise `2` to the power of `2`:

```cpp
#include <iostream>
#include <cmath>

int main(){

    double base = 2;
    double exponent = 2;

    double result = pow(base, exponent);

    std::cout << base << " raised to the power of " << exponent << " equals " << result << ".\n";

>     //Output: 2 raised to the power of 2 equals 4.
> 

Reviewer: ```suggestion
    //Output: 2 raised to the power of 2 equals 4.
```

}
```

## Codebyte Example

> The following example is runnable and uses the `pow()` function to find the `3`-root (cube root) of `27` by raising the base to the reciprocal of the desired root:

Reviewer: This example seems too similar to the [example](https://www.codecademy.com/resources/docs/cpp/math-functions/cbrt#:~:text=Use%20cbrt()%20to%20return%20the%20cube%20root%20of%2027%3A) under the existing `cbrt()` C++ entry. Wondering if we could change the values but keep the reciprocal?

```suggestion
The following example is runnable and uses the `pow()` function to find the nth-root of `16` by raising the base to the reciprocal of the desired root:
```

```codebyte/cpp
#include <iostream>
#include <cmath>

int main(){

>     double base = 27;
>     double root = 3;

Reviewer: ```suggestion
    double base = 16;
    double root = 4;
```

    double exponent = 1 / root;

    double result = pow(base, exponent);

>     std::cout << base << " raised to the power of " << exponent << " equals " << result <<", which is the " << root << "-root of " << base << ".\n";
> 

Reviewer: ```suggestion
    std::cout << base << " raised to the power of " << exponent << " equals " << result <<", which is the " << root << "-root of " << base << ".\n";
```

---
Title: 'pow()'
Description: 'Returns the result of the base raised to the power of exponent'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`pow()`** function returns the value of `base`<sup>`exponent`</sup>.

## Syntax

```cpp
pow(base, exponent)
```

## Example

The following example uses the `pow()` function to raise `2` to the power of `2`:

```cpp
#include <iostream>
#include <cmath>

int main(){

    double base = 2;
    double exponent = 2;

    double result = pow(base, exponent);

> 

Reviewer 2: ```suggestion
```

    std::cout << base << " raised to the power of " << exponent << " equals " << result << ".\n";
    //Output: 2 raised to the power of 2 equals 4.

}
```

## Codebyte Example

> The following example is runnable and uses the `pow()` function to raise `3` to the power of `2`:

Reviewer 2: Both examples are largely the same. It'd be better if you did something to differentiate the two. Maybe having one use fractional exponents to find the roots of a number:

double base = 27;
double root = 3;
double exponent = 1 / root;

That's just a suggestion, you could also do something like finding the volume of a cube or a sphere.

```codebyte/cpp
#include <iostream>
#include <cmath>

int main(){

    double base = 3;
    double exponent = 2;

    double result = pow(base, exponent);

> 

Reviewer 2: ```suggestion
```

---
Title: 'pow()'
Description: 'Returns the result of a base raised to the power of an exponent.'
Subjects:
  - 'Computer Science'
Tags:
  - 'Functions'
  - 'Arithmetic'
CatalogContent:
  - 'learn-c-plus-plus'
  - 'paths/computer-science'
---

The **`pow()`** function returns the result of a base value raised to the power of an exponent.

## Syntax

```pseudo
pow(base, exponent)
```

`base` and `exponent` are integers or of type `float`. If `base` is zero, the function returns `0.0`. If `exponent` is zero, the function returns `1.0`.

- A domain error may occur if the `base` is a float and `exponent` is not an integer value.
- A pole or range error may occur due to overflow.

> - A range error may occue due to underflow.

Reviewer 3: ```suggestion
- A range error may occur due to underflow.
```

## Example

The following example uses the `pow()` function to raise `2` to the power of `2`:

```cpp
#include <iostream>
#include <cmath>

int main()
{
  double base = 2;
  double exponent = 2;

  double result = pow(base, exponent);
  std::cout << base << " raised to the power of " << exponent << " equals " << result << ".\n";
  // Output: 2 raised to the power of 2 equals 4
}
```

## Codebyte Example

The following example is runnable and uses the `pow()` function to find the nth-root of `16` by raising the base to the reciprocal of the desired root:

```codebyte/cpp
#include <iostream>
#include <cmath>

int main()
{

>   double base = 27;

Reviewer 3: ```suggestion
  double base = 16;
```
